assets/js/common/ApiKeySettingsModal/ApiKeySettingsModal.jsx

```diff
@@ -0,0 +1,191 @@
+import React, { useEffect, useState } from 'react';
+import {
+  addDays,
+  addMonths,
+  addYears,
+  format,
+  parseISO,
+  setHours,
+  setMinutes,
+} from 'date-fns';
+import { noop, truncate } from 'lodash';
+import { EOS_CONTENT_COPY, EOS_INFO_OUTLINED } from 'eos-icons-react';
+import Button from '@common/Button';
+import Modal from '@common/Modal';
+import { InputNumber } from '@common/Input';
+import Label from '@common/Label';
+import Select from '@common/Select';
+import Switch from '@common/Switch';
+
+const normalizeExpiration = (expiration) =>
+  setMinutes(setHours(expiration, 0), 0);
+
+const availableTimeOptions = [
+  {
+    type: 'months',
+    timeGenerator: (quantity) => addMonths(new Date(), quantity),
+  },
+  {
+    type: 'days',
+    timeGenerator: (quantity) => addDays(new Date(), quantity),
+  },
+  {
+    type: 'years',
+    timeGenerator: (quantity) => addYears(new Date(), quantity),
+  },
+];
+
+function ApiKeySettingsModal({
+  open = false,
+  loading = false,
+  onGenerate = noop,
+  onClose = noop,
+  generatedApiKey,
+  generatedApiKeyExpiration,
+}) {
+  const timeOptions = availableTimeOptions.map((o) => o.type);
+  const [timeQuantity, setTimeQuantity] = useState(0);
+  const [timeQuantityType, setTimeQuantityType] = useState(timeOptions[0]);
+  const [keyGenerated, setKeyGenerated] = useState(false);
+
+  const [quantityError, setQuantityError] = useState(false);
+  const [apiKeyNeverExpires, setapiKeyNeverExpires] = useState(false);
+
+  const generateApiKeyExpiration = () => {
+    if (apiKeyNeverExpires) {
+      onGenerate({ apiKeyExpiration: null });
+      setKeyGenerated(true);
+      return;
+    }
+    if (timeQuantity === 0 || !timeQuantity) {
+      setQuantityError(true);
+      return;
+    }
+    const timeQuantitySettings = availableTimeOptions.find(
+      (q) => q.type === timeQuantityType
+    );
+    const apiKeyExpiration = timeQuantitySettings.timeGenerator(timeQuantity);
+
+    onGenerate({ apiKeyExpiration: normalizeExpiration(apiKeyExpiration) });
+    setKeyGenerated(true);
+  };
+
+  useEffect(() => {
+    setapiKeyNeverExpires(false);
+    setQuantityError(false);
+    setTimeQuantityType(timeOptions[0]);
+    setTimeQuantity(0);
+    setKeyGenerated(false);
+  }, [open]);
+
+  return (
+    <Modal
+      title="API Key Settings"
+      className="!w-3/4 !max-w-3xl"
+      open={open}
+      onClose={onClose}
+    >
+      <div className="flex flex-col my-5">
+        <span className="my-2 mb-4 text-gray-500">
+          {' '}
```

Comment:
Just curious, you had to add this empty spaces because otherwise CSS was not formatted properly, right?

Reply:
It's a prettier thingy :D